api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -254,5 +258,43 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
     public <T> Boolean notIn(BoundReference<T> ref, Literal<T> lit) {
       return ROWS_MIGHT_MATCH;
     }
+
+    @Override
+    public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+      int id = ref.fieldId();
+
+      String prefix = (String) lit.value();
+      ByteBuffer prefixAsBytes = ByteBuffer.wrap(prefix.getBytes(StandardCharsets.UTF_8));
```

Comment:
Can we cache these instead of converting in each call to evaluate? Metrics evaluators are called in a tight loop when planning jobs.

Reply:
Maybe make it so that a `StringLiteral` can cache and return the bytes representation.

Reply:
I would like to do a similar thing in `ParquetMetricsRowGroupFilter`. However, `StringLiteral` is not accessible there. Do we want to make `Literals` public?

Reply:
Maybe we should add `toByteBuffer` and `fromByteBuffer` methods to `Literal`. That would make conversion to the single-value serialization easier (and public) and would solve this problem.

Reply:
I've created #414 separately to address this.